### ISSUE TITLE
Fine-tuning pre-Trained Models

### ISSUE THREAD

**Issue by satrajitgithub** [deepmedic/deepmedic#193]
Hi,

I am trying to fine-tune a pre-train model using the command:
`./deepMedicRun -model ./examples/configFiles/deepMedic/model/modelConfig.cfg \
               -train ./examples/configFiles/deepMedic/train/trainConfigForRefinement.cfg \
               -load ./path/to/pretrained/network/filename.DATE+TIME.model.ckpt \
               -resetopt \
               -dev cuda0`
Now, I am wondering: for the -model flag, should I give the path to modelConfig of the pre-trained network or should it be the path to the new network? 

I think it is the path to the modelConfig of the new network and the only thing I am using from the pretrained network is `-load <pretrained_model>`  but I still wanted to confirm this. 

Thanks in advance.

**Comment by Kamnitsask** [comment]
Hello,

Arent the two architectures the same? In which case, the modelConfig would be the same, and only differences would be in the trainConfig (eg different database or different learning schedule)?

I am not sure if the current code will work for the case that the architecture changes (eg adding a layer over the pretrained). In this case, tensorflow currently would not know which weights correspond to what layer I think. A custor weight loader would need to be created.

So, normal use-cases would keep the same modelConfig. If the only thing that changes is something that does not add/remove trainable parameters (eg activation functions or dropout etc), then the loading should work with the new modelConfig.

I hope the above helps. Feel free to ask further if you are still unclear.

Kind regards,
Kostas

**Comment by satrajitgithub** [comment]
Hi Kostas,

Thanks for your clarification. After posting the issue, I realized that the two models indeed need to have the same architecture. If I use a different modelConfig with different parameters (say number of classes), it says graph doesn't match (which is expected). 

Regards,
Satrajit
